language/diem-framework/modules/DiemVMConfig.move

```diff
@@ -134,6 +138,79 @@ module DiemVMConfig {
             }};
     }
 
+    public fun set_gas_constants(
+        dr_account: &signer,
+        global_memory_per_byte_cost: u64,
+        global_memory_per_byte_write_cost: u64,
+        min_transaction_gas_units: u64,
+        large_transaction_cutoff: u64,
+        intrinsic_gas_per_byte: u64,
+        maximum_number_of_gas_units: u64,
+        min_price_per_gas_unit: u64,
+        max_price_per_gas_unit: u64,
+        max_transaction_size_in_bytes: u64,
+        gas_unit_scaling_factor: u64,
+        default_account_size: u64,
+    ) {
+        DiemTimestamp::assert_operating();
+        Roles::assert_diem_root(dr_account);
+        assert(
+            min_price_per_gas_unit <= max_price_per_gas_unit,
+            Errors::invalid_argument(EGAS_CONSTANT_INCONSISTENCY)
+        );
+        assert(
+            min_transaction_gas_units <= maximum_number_of_gas_units,
+            Errors::invalid_argument(EGAS_CONSTANT_INCONSISTENCY)
+        );
+
+        let config = DiemConfig::get<DiemVMConfig>();
+        let gas_constants = &mut config.gas_schedule.gas_constants;
+
+        gas_constants.global_memory_per_byte_cost       = global_memory_per_byte_cost;
+        gas_constants.global_memory_per_byte_write_cost = global_memory_per_byte_write_cost;
+        gas_constants.min_transaction_gas_units         = min_transaction_gas_units;
+        gas_constants.large_transaction_cutoff          = large_transaction_cutoff;
+        gas_constants.intrinsic_gas_per_byte            = intrinsic_gas_per_byte;
+        gas_constants.maximum_number_of_gas_units       = maximum_number_of_gas_units;
+        gas_constants.min_price_per_gas_unit            = min_price_per_gas_unit;
+        gas_constants.max_price_per_gas_unit            = max_price_per_gas_unit;
+        gas_constants.max_transaction_size_in_bytes     = max_transaction_size_in_bytes;
+        gas_constants.gas_unit_scaling_factor           = gas_unit_scaling_factor;
+        gas_constants.default_account_size              = default_account_size;
+
+        DiemConfig::set(dr_account, config);
+    }
+    spec fun set_gas_constants {
+        include DiemTimestamp::AbortsIfNotOperating;
+        /// No one can update DiemVMConfig except for the Diem Root account [[H11]][PERMISSION]
```

Comment:
This follows the permission module in DIP-2, cc @junkil-park just to double-check/FYI as you're auditing this at the moment.

Reply:
LGTM. Nit: could you add a period "." at the end of the sentence?